flink-libraries/flink-table/src/main/java/org/apache/flink/table/plan/nodes/datastream/aggs/DoubleSummaryAggregation.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.plan.nodes.datastream.aggs;
+
+import static org.apache.flink.api.java.summarize.aggregation.CompensatedSum.ZERO;
+
+import org.apache.flink.api.java.summarize.aggregation.Aggregator;
+import org.apache.flink.api.java.summarize.aggregation.CompensatedSum;
+import org.apache.flink.api.java.summarize.aggregation.NumericSummaryAggregator;
+
+public class DoubleSummaryAggregation extends NumericSummaryAggregator<Double> {
```

Comment:
@huawei-flink ,   I have replied your question in the UDAGG design doc. AggregateFunction is the base class for UDAGG. We are very cautious to add any new method into this interface. As mentioned in the UDAGG design doc, only createAccumulator, getValue, accumulate are the must to have methods for an aggregate. Merge methods is optional only useful for advanced optimization for the runtime execution plan. Retract may also be a must-have if the users are care about the correctness. I do not see why reset is necessary for aggregate. If it is helpful in your case, you can always add this method in your User(you as the user) Defined Aggregate Function. UDAGG is still on the way, but I think it should be available very soon.